nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-pubsub-processors/src/test/java/org/apache/nifi/processors/kafka/pubsub/AbstractKafkaProcessorLifecycelTest.java

```diff
@@ -0,0 +1,253 @@
+            assertTrue(e.getMessage().contains("'topic' is invalid because topic is required"));
+        }
+
+        runner.setProperty(ConsumeKafka.TOPIC, "");
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("must contain at least one character that is not white space"));
+        }
+
+        runner.setProperty(ConsumeKafka.TOPIC, "  ");
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("must contain at least one character that is not white space"));
+        }
+        runner.setProperty(ConsumeKafka.TOPIC, "blah");
+
+        runner.removeProperty(ConsumeKafka.CLIENT_ID);
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("invalid because client.id is required"));
+        }
+
+        runner.setProperty(ConsumeKafka.CLIENT_ID, "");
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("must contain at least one character that is not white space"));
+        }
+
+        runner.setProperty(ConsumeKafka.CLIENT_ID, "   ");
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("must contain at least one character that is not white space"));
+        }
+        runner.setProperty(ConsumeKafka.CLIENT_ID, "ghj");
+
+        runner.setProperty(PublishKafka.KERBEROS_PRINCIPLE, "");
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("must contain at least one character that is not white space"));
+        }
+        runner.setProperty(PublishKafka.KERBEROS_PRINCIPLE, "  ");
+        try {
+            runner.assertValid();
+            fail();
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("must contain at least one character that is not white space"));
+        }
+    }
+
+    /*
+     * The goal of this test is to validate the correctness of
+     * AbstractKafkaProcessor's implementation of onTrigger() in a highly
+     * concurrent environment. That is:
+     * - upon processing failures (e.g., unhandled exceptions), the target Kafka
+     *   resource is reset (closed and re-created)
+     * - no FlowFile is unaccounted for. FlowFiles left in the queue and FlowFiles
+     *   in Success relationship = testCount
+     * - failed executions that did not result in the call to close/reset summed with
+     *   verified calls to close should equal total request failed
+     */
+    @Test
+    public void validateLifecycleCorrectnessWithProcessingFailures() throws Exception {
```

Comment:
It seems that -D< property >=< value > where property contains "." is not allowed. I tried with -Dencoding=UTF-8 and no change. Then I found:
http://stackoverflow.com/questions/17656475/maven-source-encoding-in-utf-8-not-working

I added in nifi-kafka-pubsub-processors\pom.xml
````xml
	<build>
		<plugins>
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-surefire-plugin</artifactId>
				<configuration>
					<argLine>-Dfile.encoding=UTF-8</argLine>
				</configuration>
			</plugin>
		</plugins>
	</build>
````

And it is now OK. Maybe this should be placed in a parent pom file?